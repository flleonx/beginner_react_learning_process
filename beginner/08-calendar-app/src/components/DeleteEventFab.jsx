import { useDispatch } from 'react-redux';

import { calendarEventStartDelete } from '../actions/calendarEvent';

export const DeleteEventFab = () => {

  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(calendarEventStartDelete());
  };

  return (
    <button
      className="btn btn-danger fab-danger"
      onClick={handleDelete}
    >
      <i className="fas fa-trash"></i>
      <span> Eliminar evento</span>
    </button>
  )
};
