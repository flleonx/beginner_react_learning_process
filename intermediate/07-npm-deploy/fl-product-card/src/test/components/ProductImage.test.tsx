import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../components';
import { product2 } from '../data/products';

describe('ProductImage', () => {

  test('It should show <ProductImage /> properly with the custom image', () => {
    const wrapper = renderer.create(
      <ProductImage img="test-image.png"/>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('It should show the component with the product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

});
