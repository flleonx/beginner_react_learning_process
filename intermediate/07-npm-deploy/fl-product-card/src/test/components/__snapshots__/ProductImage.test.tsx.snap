// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage It should show <ProductImage /> properly with the custom image 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="test-image.png"
/>
`;

exports[`ProductImage It should show the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
